code/modules/hydroponics/grown/corn.dm

```diff
@@ -89,5 +90,25 @@
 		user.put_in_hands(S)
 	snap_pops -= 1
 	if(!snap_pops)
-		new /obj/item/grown/corncob(user.loc)
+		new /obj/item/grown/corncob/snap(user.loc)
 		qdel(src)
+
+/obj/item/grown/corncob/snap
+	seed = /obj/item/seeds/corn/snapcorn
+	name = "snap corn cob"
+	desc = "A reminder of pranks gone by."
+	icon_state = "corncob"
+	inhand_icon_state = null
+	w_class = WEIGHT_CLASS_TINY
+	throwforce = 0
+	throw_speed = 3
+	throw_range = 7
+	grind_results = list(/datum/reagent/cellulose = 10) //really partially hemicellulose
+
+/obj/item/grown/corncob/snap/attackby(obj/item/grown/W, mob/user, params)
+	if(W.get_sharpness())
+		to_chat(user, span_notice("You use [W] to fashion a pipe out of the corn cob!"))
+		new /obj/item/clothing/mask/cigarette/pipe/cobpipe (user.loc)
+		qdel(src)
+	else
+		return ..()
```

Comment:
All of this should no longer be necessary since you are inheriting it form your parent
```suggestion
```